influxdata/influxdb
# feat(ui): added auth0 to branch and added form functionality

This is part two of a larger epic that is integrating auth0 sign-up and log-in capabilities for cloud users on the region they are trying to access. This PR adds a lot of the meat and potatoes for this functionality. Auth0 is added as a dependency to allow users to directly sign-up & log-in through the newly created `login` route & LoginPage. This PR allows users to toggle between the sign-up form and the log-in form, and request to change their password. In addition to the form logic that was built out, this PR also adds a new shared component called `SocialButton` that allows developers to pass button-specific properties (button text, click functionality, and children) to reduce repetition when adding new social sign-up capabilities. Finally, two SVGs were added (github and google) to allow for customized buttons.

**HOW TO TEST**

In order to test this functionality, you will need to run the UI as a cloud-dev user, and enable the feature flag. You will need to replace the current auth0js config variables with valid config variables that can be obtained from Engineering Vault quartz/.env file (auth0 clientID, domain). The redirectURL should be whichever home route your local development is happening on (i.e. localhost:8080/, etc...).
